### ISSUE TITLE
"Must select a database"

### ISSUE THREAD

**Issue by wvmikep** [project-open-data/db-to-api#14]
I set up the config.php as instructed and I'm getting "Must select a database."  Is there nothing else I need to edit?

**Comment by layer67tech** [comment]
Yes. You will need to edit the class.db-api.php file in the include folder. This is the manual process.
    /**
     \* Register a new dataset
     \* @param string $name the dataset name
     \* @param array $args the dataset properties
     */
    function register_db( $name = null, $args = array() ) {

```
    $defaults = array(
        'name' => null,
        'username' => 'root',
        'password' => 'root',
        'server' => 'localhost',
        'port' => 3306,
        'type' => 'mysql',
        'table_blacklist' => array(),
        'column_blacklist' => array(),
        'ttl' => $this->ttl,
    );

    $args = shortcode_atts( $defaults, $args );
    $name = $this->slugify( $name );

    $this->dbs[$name] = (object) $args;

}
```

**Comment by wvmikep** [comment]
It doesn't help at all.  Same result.  I know the info I give it is good because I have a separate test page that returns data.

Trying to reverse engineer this, starting  with shortcode_atts.

**Comment by layer67tech** [comment]
When you received the error message "Select a database", this means your config.php and class.db-api.php files have not been configured correctly. what do you have in the config.php and class.db-api files?

**Comment by wvmikep** [comment]
They're below.  

With that said, it's failing in parse_query because there's no query string.  If I put db and table in the query string (not intended per the instructions ... and doing so makes no sense anyway), I then get invalid database.

Doesn't make sense given how my little test page returns data from the db with the same credentials.

# 

$args = array( 
            'name' => 'dbname',
            'username' => 'dbuser',
            'password' => 'dbpw',
            'server' => 'localhost',
            'port' => 3306,
            'type' => 'mysql',
            'table_blacklist' => array(),
            'column_blacklist' => array(),
);

register_db_api( 'berkeleywv-fe-inspections', $args );

# 

```
function register_db( $name = null, $args = array() ) {

    $defaults = array(
        'name' => 'dbname',
        'username' => 'dbuser',
        'password' => 'dbpw',
        'server' => 'localhost',
        'port' => 3306,
        'type' => 'mysql',
        'table_blacklist' => array(),
        'column_blacklist' => array(),
        'ttl' => $this->ttl,
    );

    $args = shortcode_atts( $defaults, $args );
    $name = $this->slugify( $name );

    $this->dbs[$name] = (object) $args;

}
```

**Comment by layer67tech** [comment]
This appears to be a database and server access issue. Test your sample db_user , db_pass, db_host and db_name in a separate SQL Admin tool to see if you can access the database server.

**Comment by wvmikep** [comment]
That "test page" is separate from db-to-API. Phpmyadmin has no issues either.

**Comment by layer67tech** [comment]
okay, try accessing one of your tables in the database.
e.g.: http://localhost/berkeleywv-fe-inspections/table.json
e.g.: http://localhost/berkeleywv-fe-inspections/table.xml
e.g.: http://localhost/berkeleywv-fe-inspections/table.html

**Comment by wvmikep** [comment]
Must select a database

**Comment by layer67tech** [comment]
Absolutely a database connection to web server issue. This is usually resolved by changing your database binding in the my.cnf configuration.
e.g.: #bind-address           = 127.0.0.1

**Comment by wvmikep** [comment]
Not so fast: http://data.mike-pulsifer.org/test.php

**Comment by layer67tech** [comment]
Problem with code. Does not work that way. test.php cannot be converted to REST
http://avizium.no-ip.org/apiv2/avizium-bloggers/av_comments.xml

**Comment by wvmikep** [comment]
I know. It just proves I can get to the db.

**Comment by phoenixg** [comment]
Have you solved that? I meet the same problem.

**Comment by waldoj** [comment]
FWIW, I've been unable to replicate the problem.

**Comment by jamespsterling** [comment]
@alpheustechnologies You're saying we have to edit the config.php and class.db-api.php. Why have a config file then?

**Comment by bunlongheng** [comment]
@I faced the same problem. 

I've followed all the comments in this post, and still see 

> Must select a database

---

I've used the same settings in MySQL Workbench and I can connect to my db fine. 

---

What should I put for **dataset-name** ?

**Comment by ersoviet** [comment]
same problem!

**Comment by MelanieFirefly** [comment]
You do not need to edit class.db-api.php. You need to edit config.php and .htaccess to fit your configuration:

config.php:
change $defaults = array to your db info
register_db_api( 'whateveryouwanttonameyourdataset', $args );

.htaccess:
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /folderyoumadeforthisproject/

then to access whateveryouwanttonameyourdataset dataset go to yourdomain.com/folderyoumadeforthisproject/whateveryouwanttonameyourdataset/yourtablename.json

**Comment by Anymfah** [comment]
.htaccess RewriteBase was wrong for me :) Changed it to my project path, ty man.
